C4InterFlow.Cli/Properties/launchSettings.json

```diff
@@ -1,8 +1,8 @@
 {
   "profiles": {
     "C4InterFlow.Cli": {
```

Comment:
Could we add launch profiles for draw-diagrams command for all samples as follows:
- "E-Commerce Platform - YAML - Draw Diagrams" - Samples/E-Commerce Platform/Yaml
- "Internet Banking System - YAML - Draw Diagrams" - Samples/Internet Banking System/Yaml
- "Internet Banking System - CSV/YAML - Draw Diagrams" - Samples/Internet Banking System/CSV/Architecture/Yaml
- "Internet Banking System - CSV/JSON - Draw Diagrams" - Samples/Internet Banking System/CSV/Architecture/Json
- "ToDoApp - YAML - Draw Diagrams" - Samples/ToDoApp
- "TraderX - CSV/YAML - Draw Diagrams" - Samples/TraderX/CSV/Architecture/Yaml
- "TraderX - CSV/JSON - Draw Diagrams" - Samples/TraderX/CSV/Architecture/Json
- "DotNetEShop - YAML - Draw Diagrams" - Samples/dotnet.eShop/DotNetEShop/Yaml
- "DotNetEShop - C# - Draw Diagrams" - Samples/dotnet.eShop/DotNetEShop

Later we can also add launch profiles for other commands e.g. execute-aac-strategy, publish-site etc.
Then we can get rid of CLI Commands.txt as I was mainly using it to keep multiple commands for multiple samples, so I can just copy/paste.

Reply:
Will do. I kind of figured that was the purpose of that document.

That's also why I was thinking a "unit" test project that basically has tests that look like

```csharp
[Test]
public async Task CanRunDrawCommand() {
  var commandArgs = string.join(" ", [
    "draw-diagrams",
    "--interfaces ..SoftwareSystems.*.Interfaces.*",
    "--business-processes ..BusinessProcesses.*",
    "--level-of-details context container",
    ...
  ]);

  var rootCommandBuilder = RootCommandBuilder
    .CreateDefaultBuilder(commandArgs)
    .Configure(c => c.Add<DrawDiagramsCommand>());

  var errorCode = await new CommandLineBuilder(rootCommandBuilder.Build())
    .UseDefaults()
    .UseLogging()
    .Build()
    .InvokeAsync(commandArgs);

  Assert.IsEqual(0, errorCode);
}
```

Reply:
I'm not quite sure if this kind of tests will add much value i.e. all it tells us is that command execution did not fail, but what is important is the output.

If we had automated tests then I think they could be around certain features e.g. AaC validation, resolving aliases to structures etc.

We may also have end-to-end tests (e.g. BDD/SpecFlow), like GIVEN certain AaC WHEN certain commands are executed, THEN we'd expect certain diagrams to be generated with specific PlantUML content.

The issue with this approach though is that there are 1000s of combinations of arguments (i.e. Scope, Level of Details, Diagram Types etc.), which would make it very difficult and time consuming to reach a reasonable degree of coverage. 

For now the testing that I'm doing feels enough as by running through all .bat files for all samples has proven to pick up  issues because of the samples diversity. It is time-consuming though, so maybe that can be automated somehow (?)